oshi-core/src/main/java/oshi/software/os/mac/AbstractMacOperatingSystem.java

```diff
@@ -0,0 +1,179 @@
+package oshi.software.os.mac;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import oshi.annotation.concurrent.ThreadSafe;
+import oshi.software.common.AbstractOperatingSystem;
+import oshi.software.os.ApplicationInfo;
+import oshi.software.os.OSProcess;
+import oshi.software.os.OSService;
+import oshi.software.os.OSThread;
+import oshi.util.ExecutingCommand;
+import oshi.util.FileUtil;
+import oshi.util.Memoizer;
+import oshi.util.ParseUtil;
+import oshi.util.Util;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static oshi.software.os.OSService.State.RUNNING;
+import static oshi.software.os.OSService.State.STOPPED;
+import static oshi.util.Memoizer.installedAppsExpiration;
+
+/**
+ * macOS, previously Mac OS X and later OS X) is a series of proprietary graphical operating systems developed and
+ * marketed by Apple Inc. since 2001. It is the primary operating system for Apple's Mac computers.
+ */
+@ThreadSafe
+public abstract class AbstractMacOperatingSystem extends AbstractOperatingSystem {
```

Comment:
Wouldn't it be cleaner to keep the existing `MacOperatingSystem` inheritance and just extract any JNA-related things to a new `MacOperatingSystemJNA` subclass?

Reply:
Sounds good to me. The only change I see in my code would be that I'd rename `AbstractMacOperatingSystem -> MacOperatingSystem` and `MacOperatingSystem` -> `MacOperatingSystemJNA`, right?